.circleci/config.yml

```diff
@@ -0,0 +1,41 @@
+version: 2
+jobs:
+  build:
+    docker:
+      - image: circleci/node:8
+
+    working_directory: ~/insight-website
+
+    steps:
+      - checkout
+
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "package.json" }}
+            - v1-dependencies
+
+      - run:
+          name: "Install dependencies"
+          command: yarn install
+
+      - save_cache:
+          paths:
+            - node_modules
+          key: v1-dependencies-{{ checksum "package.json" }}
+
+      - run:
+          name: "Build"
+          command: yarn build
+
+      - run:
+          name: "Netlify deploy"
+          command: |
+            if [[ "$CIRCLE_BRANCH" == "master" ]]; then
+              npm run netlify:deploy
+            fi
```

Comment:
You can filter the branch you want this to run on by using workflows, see: https://circleci.com/docs/2.0/configuration-reference/#branches